pkg/datapath/loader/netlink.go

```diff
@@ -391,12 +396,18 @@ func (l *Loader) reloadIPSecOnLinkChanges() {
 				select {
 				case <-settled:
 					break settleLoop
-				case <-updates:
+				case u := <-updates:
```

Comment:
I find the settleLoop block hard to read. If I understand it correctly I think this would do the equivalent:
```golang
startTime := time.Now()
for time.Since(startTime) < settleDuration {
	select {
	case u := <-updates:
		// Ignore veth devices
		if u.Type() == "veth" {
			continue
		}

		// Trigger reinit immediately after
		// settle duration has passed.
		select {
		case trigger <- struct{}{}:
		default:
		}
	default:
	}
}
```
Probably opinionated, but without the `settleLoop` label I find it easier to understand.

Reply:
Yeah this looks nicer. I'll refactor it to something akin to this.

Reply:
Ah the semantics are not the same. Doing a for loop will get stuck if there's no updates, so we must do a select on `After` channel. Since I've already tested this code and the reason for this PR was last minute changes, I'm going to skip changing this :)